src/components/AddressItem/AddressesItem.js

```diff
@@ -0,0 +1,91 @@
+import { useState, useRef, useEffect } from 'react';
+import ButtonRound from 'components/UI/ButtonRound';
+import PropTypes from 'prop-types';
+import { toast } from 'react-toastify';
+import { useDeleteAddressMutation } from 'utils/rtk-query-addresses';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrashCan } from '@fortawesome/free-regular-svg-icons';
+import { useUpdateAddressMutation } from 'utils/rtk-query-addresses';
+
+const AddressesItem = ({ id, address }) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [addressValue, setAddressValue] = useState(address);
+  const inputRef = useRef();
+
+  const [deleteAddress] = useDeleteAddressMutation();
+  const [updateAddress] = useUpdateAddressMutation();
+
+  const iconDelete = <FontAwesomeIcon icon={faTrashCan} />;
+
+  useEffect(() => {
+    inputRef?.current?.focus();
+  }, [isEditing]);
+
+  const deleteAddressHandler = async () => {
+    try {
+      const response = await deleteAddress(id);
+      if (response.error) {
+        throw new Error();
+      }
+    } catch (err) {
+      toast.error(
+        `We could'n delete address: ${address}. Try again later or contact us please!`
+      );
+    }
+  };
+
+  const changeInputTypeHandler = () => {
+    setIsEditing(true);
+  };
+
+  const editAddressHandler = (e) => {
+    setAddressValue(e.target.value);
+  };
+
+  const sendEditRequestHandler = () => {
+    updateAddress({ id: id, address: addressValue })
+      .unwrap()
```

Comment:
Why is it necessary?

Reply:
Without *.unwrap()* in case that some error appear, catch() does'n see it. I checked it and there is info about it in documentation: https://redux-toolkit.js.org/rtk-query/usage/error-handling#error-display-examples